packages/nx/src/command-line/migrate.ts

```diff
@@ -158,146 +175,137 @@ export class Migrator {
+        .map((packageName) =>
+          this._updatePackageJson(packageName, packages[packageName], {
             ...collectedVersions,
             [targetPackage]: target,
           })
         )
     );
-    return childCalls.reduce(
-      (m, c) => {
-        Object.keys(c).forEach((r) => {
-          if (!m[r] || this.gt(c[r].version, m[r].version)) {
-            m[r] = c[r];
+
+    return childPackageMigrations.reduce(
+      (migrations, childMigrations) => {
+        for (const migrationName of Object.keys(childMigrations)) {
+          if (
+            !migrations[migrationName] ||
+            this.gt(
+              childMigrations[migrationName].version,
+              migrations[migrationName].version
+            )
+          ) {
+            migrations[migrationName] = childMigrations[migrationName];
           }
-        });
-        return m;
+        }
+        return migrations;
       },
       {
         [targetPackage]: {
           version: migrationsJson.version,
           addToPackageJson: target.addToPackageJson || false,
         },
-      }
+      } as Record<string, PackageJsonUpdateForPackage>
     );
   }
 
   private collapsePackages(
     packageName: string,
     targetVersion: string,
-    m: MigrationsJson | null
-  ) {
+    migration: ResolvedMigrationConfiguration
+  ): Record<string, PackageJsonUpdateForPackage> {
     // this should be used to know what version to include
     // we should use from everywhere we use versions
 
-    if (packageName === '@nrwl/workspace') {
-      if (!m.packageJsonUpdates) m.packageJsonUpdates = {};
-      m.packageJsonUpdates[`${targetVersion}-defaultPackages`] = {
+    if (migration.packageGroup) {
+      migration.packageJsonUpdates ??= {};
+      migration.packageJsonUpdates[`${targetVersion}-defaultPackages`] = {
         version: targetVersion,
-        packages: [
-          'nx',
-          '@nrwl/angular',
-          '@nrwl/cypress',
-          '@nrwl/devkit',
-          '@nrwl/eslint-plugin-nx',
-          '@nrwl/express',
-          '@nrwl/jest',
-          '@nrwl/js',
-          '@nrwl/cli',
-          '@nrwl/linter',
-          '@nrwl/nest',
-          '@nrwl/next',
-          '@nrwl/node',
-          '@nrwl/nx-cloud',
-          '@nrwl/nx-plugin',
-          '@nrwl/react',
-          '@nrwl/storybook',
-          '@nrwl/web',
-          '@nrwl/react-native',
-          '@nrwl/detox',
-        ].reduce(
-          (m, c) => ({
-            ...m,
-            [c]: {
-              version: c === '@nrwl/nx-cloud' ? 'latest' : targetVersion,
+        packages: migration.packageGroup.reduce((acc, packageConfig) => {
+          const { package: pkg, version } =
+            typeof packageConfig === 'string'
+              ? { package: packageConfig, version: targetVersion }
+              : packageConfig;
+
+          return {
+            ...acc,
+            [pkg]: {
+              version,
               alwaysAddToPackageJson: false,
```

Comment:
Should we allow this part to be specc'd from the packageGroup object? Imagine a plugin introduces a new utils package, should they have to write a migration that installs it or should they be able to add it to packageGroup with `alwaysAddToPackageJson` set to true?

Reply:
If I allow `version` (to support `nx-cloud`), I can definitely allow the rest of the keys if wanted.